src/sync/app.cpp

```diff
@@ -73,10 +73,23 @@ App::App(const Config& config)
 
 static Optional<AppError> check_for_errors(const Response& response)
 {
+    auto http_status_code_is_fatal = [response] {
+        if (response.http_status_code >= 300
+            || (response.http_status_code < 200 && response.http_status_code != 0)) {
+            return true;
+        }
+        return false;
+    };
+    
     try {
-        if (auto ct = response.headers.find("Content-Type"); ct != response.headers.end() && ct->second == "application/json") {
+        if (auto ct = response.headers.find("Content-Type");
```

Comment:
Having the wrapped part of the if indented to the same level as the following block makes it hard to tell where the body of the if block starts. When the condition for an `if` statement is too big to reasonable fit on a line it's usually better to extract it to a local variable or helper function.

Reply:
I agree, I've cleaned it up to be more readable